design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,471 @@
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | Allows specific identification of the Secrets Provider **Job**, which enhances security | \- Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider<br />- ***Not*** a ***seamless*** experience between Milestones because i | 10 days           |
+
+Implementing the Job granularity would mean that the experience in transitioning between the Milestones is not a smooth one. If a customer uses the Job granularity, they will need to update this in Milestone 2 because the deployment type will no longer be a Job. 
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy. We would recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer experience
+
+Deployment for the Secrets Provider will be done using Helm. That way, if the 
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined
+
+2. Create `values.yaml` that will contain the following parameters and their defaults:
+
+   For a full list of required parameters, see `values.yaml` below
+
+   *When parameters are not configured by customer, their defaults are used*
+
+3. Install Helm Chart for the Secrets Provider
+
+We will supply the customer with a  `values.yaml` file where all default parameters are collected. 
+
+*values.yaml*
+
+```yaml
+global:
+  namespace: default
+	
+# K8s permission defaults
+rbac:
+  serviceAccountName: secrets-provider-service-account
+  roleName: secrets-provider-role
+  roleBinding: secrets-provider-role-binding
+
+# Secrets Provider pod defaults
+secrets-provider:
+  image: cyberark/secrets-provider-for-k8s
+  name: cyberark-secrets-provider
+	
+# Secrets Provider environment variable defaults
+secrets-provider-environment:
+  conjurSslCertificateName: secrets-provider-ssl-config-map
+  containerMode: init
```

Comment:
Don't we want to have the use case as the parameter? so instead of `containerMode` we can have something like `serveMultipleApps: false` for init

Reply:
containerMode is value that comes from the authn client. We didn't want to change the containerMode value for now because the job is still acting like an init container where, although not attached to a certain application, it runs before all the pods, spins up to perform it's task and terminates upon completion

Reply:
i'll just clarify something - `containerMode` is set in the config of the authn-client but is used in the secrets-provider for terminating the container after one cycle if it is set to `init`. If we don't provide it, and instead query on some other variable then everything will be fine. 

I tried to hint in that direction in another comment i gave in this doc. I think that this is the correct approach. `container_mode = init` is pretty weird when the container is not an init container. 

However, i don't think that `serveMultipleApps` will give us what you are looking for as this is not the question we want to ask when we terminate the pod or wait for another cycle. i think that `performRoatation` (or something like that) will be more fitting and will give what you need. this way you can remove the `containerMode`.

Reply:
this is the other thread - https://github.com/cyberark/secrets-provider-for-k8s/pull/151#discussion_r454826482

Reply:
> If we don't provide it, and instead query on some other variable then everything will be fine.

Are you saying that if we another value and it will pass even if it was `containerMode=foobar` and we are only checking for init?

Reply:
correct. I encourage you to look in the code of the secrets-provider and the authn-client to see for yourself :)

Reply:
or just add the environment variable `performRotation=false` in the manifest and check for that and add in code
```
if authnConfig.ContainerMode == "init" || authncConfig.performRotation=="false" {
     os.Exit(0)
}
```

Reply:
yes, that's exactly what I was thinking about